crates/vim/src/motion.rs

```diff
@@ -1944,6 +1970,26 @@ pub(crate) fn start_of_line(
     }
 }
 
+pub(crate) fn middle_of_line(
+    map: &DisplaySnapshot,
+    display_lines: bool,
+    point: DisplayPoint,
+) -> DisplayPoint {
+    if display_lines {
+        map.clip_point(
+            DisplayPoint::new(point.row(), map.line_len(point.row()) / 2),
+            Bias::Left,
+        )
+    } else {
+        let mut mid = map.prev_line_boundary(point.to_point(map)).1;
```

Comment:
I think we need to calculate this in "buffer" co-ordinates and convert back. Something like:

````
let mut buffer_point = point.to_point(map);
buffer_point.column = map.buffer_snapshot.line_len(MultiBufferRow(buffer_point.row)) / 2
buffer_point.to_display_point(&map, Bias::Left)
```